backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/TestingSessionTest.java

```diff
@@ -0,0 +1,232 @@
+import org.apache.james.backends.cassandra.versions.SchemaVersion;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+class TestingSessionTest {
+    @RegisterExtension
+    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraSchemaVersionModule.MODULE);
+
+    private CassandraSchemaVersionDAO dao;
+
+    @BeforeEach
+    void setUp(CassandraCluster cassandra) {
+        dao = new CassandraSchemaVersionDAO(cassandra.getConf());
+    }
+
+    @Test
+    void daoOperationShouldNotBeInstrumentedByDefault() {
+        assertThatCode(() -> dao.getCurrentSchemaVersion().block())
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void daoOperationShouldNotBeInstrumentedWhenNotMatching(CassandraCluster cassandra) {
+        cassandra.getConf()
+            .fail()
+            .whenBoundStatementStartsWith("non matching")
+            .times(1)
+            .setExecutionHook();
+
+        assertThatCode(() -> dao.getCurrentSchemaVersion().block())
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void daoOperationShouldNotBeInstrumentedWhenTimesIsZero(CassandraCluster cassandra) {
+        cassandra.getConf()
+            .fail()
+            .whenBoundStatementStartsWith("SELECT value FROM schemaVersion;")
+            .times(0)
+            .setExecutionHook();
+
+        assertThatCode(() -> dao.getCurrentSchemaVersion().block())
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void daoOperationShouldNotBeInstrumentedWhenTimesIsNegative(CassandraCluster cassandra) {
+        cassandra.getConf()
+            .fail()
+            .whenBoundStatementStartsWith("SELECT value FROM schemaVersion;")
+            .times(-1)
+            .setExecutionHook();
+
+        assertThatCode(() -> dao.getCurrentSchemaVersion().block())
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void daoOperationShouldFailWhenInstrumented(CassandraCluster cassandra) {
+        cassandra.getConf()
+            .fail()
+            .whenBoundStatementStartsWith("SELECT value FROM schemaVersion;")
+            .times(1)
+            .setExecutionHook();
+
+        assertThatThrownBy(() -> dao.getCurrentSchemaVersion().block())
+            .isInstanceOf(RuntimeException.class);
+    }
+
+    @Test
+    void daoShouldNotBeInstrumentedWhenTimesIsExceeded(CassandraCluster cassandra) {
+        cassandra.getConf()
+            .fail()
+            .whenBoundStatementStartsWith("SELECT value FROM schemaVersion;")
+            .times(1)
+            .setExecutionHook();
+
+        try {
+            dao.getCurrentSchemaVersion().block();
+        } catch (Exception e) {
+            // discard expected exception
```

Comment:
assert it

Reply:
As far as I know this is what `daoOperationShouldFailWhenInstrumented` tests.

Asserting it will make the current test case less clean as I will have several assertions.